File: Inherent/002.rb
#계산기 만들기
class Cal
  def initialize(v1, v2)
    @v1 = v1
    @v2 = v2
  end

  def add()
    return @v1+@v2
  end

  def setV1(v)
    if v.class()==Integer #복잡하게 해보기
      @v1 = v
    else
      puts("Integer가 아님")
    end
  end

  def setV2(v)
    if v.is_a?(Integer)
      @v2 = v
    else
      puts("Integer가 아님")
    end
  end

  def getV1()
    return @v1
  end

  def getV2()
    return @v2
  end

end

c1 = Cal.new(10,10)
c1.setV1(5)
p (c1.getV1)
c1.setV1('10')
p (c1.getV2)

File: Basic/OOP_Variable/002.py
#계산기 만들기
class Cal(object):
    def __init__(self, v1, v2):
        # prv(self)
        # print(v1, v2)
        self.v1 = v1
        self.v2 = v2
    def add(self):
        return self.v1+self.v2
    def substract(self):
        return self.v1-self.v2
    def setV1(self, v):
        # dataType = type(v)
        # print(dataType)
        # if dataType == "<class 'int'>":
        if isinstance(v, int):
            self.v1 = v
            print(self.v1, "은 이제 self.v1")
        # elif isinstance(v, str):
        #     print("str, 문자열이네여")
        else:
            print("int도 str가 아닙니다.")

    def setV2(self, v):
        if isinstance(v, int):
            self.v2 = v
            print(self.v2, "은 이제 self.v2")
        else:
            print("int가 아닙니다.")

    def getV1(self, v):
        return self.v1
    def getV2(self, v):
        return self.v2

c1 = Cal(10,10)

c1.setV1(5)
c1.setV1('5')

c1.v2 = 7
